coalib/parsing/Filter.py

```diff
@@ -0,0 +1,19 @@
+from coalib.parsing.FilterImplementation import *
+
+
+class Filter:
+    all_filters = {'language': LanguageFilter}
+
+    def __init__(self):
+        pass
+
+    def is_valid_filter(filter):
+        return filter in Filter.all_filters
+
+    @staticmethod
+    def get_filtered_bears(filter, args, log_printer):
+        if Filter.is_valid_filter(filter):
+            return Filter.all_filters[filter](args, log_printer)
```

Comment:
what about passing each filter the list of bears? :) Then the filters don't have to look for them themselves, but we have one function collecting them all, and then each filter can process them :)